x-pack/legacy/plugins/apm/server/lib/metrics/fetch_and_transform_metrics.ts

```diff
@@ -45,23 +42,16 @@ export async function fetchAndTransformMetrics<T extends Aggs>({
   aggs: T;
   additionalFilters?: Filter[];
 }) {
-  const { start, end, uiFiltersES, client, config } = setup;
+  const { start, end, client } = setup;
 
-  const params = {
-    index: config.get<string>('apm_oss.metricsIndices'),
+  const projection = getMetricsProjection({ setup, serviceName });
+
+  const params = mergeProjection(projection, {
     body: {
       size: 0,
       query: {
         bool: {
-          filter: [
-            { term: { [SERVICE_NAME]: serviceName } },
-            { term: { [PROCESSOR_EVENT]: 'metric' } },
-            {
-              range: rangeFilter(start, end)
-            },
-            ...additionalFilters,
-            ...uiFiltersES
-          ]
+          filter: [...projection.body.query.bool.filter, ...additionalFilters]
```

Comment:
Shouldn't this be handled by `mergeProjection`?

Reply:
I was considering having mergeProjection concatenate arrays, but I opted against it because I thought the semantics would be more clear (only merges plain objects). If we concatenate arrays it will make it hard for instance to empty an array. Bit of a 50/50 one for me though. What do you think?

Reply:
> If we concatenate arrays it will make it hard for instance to empty an array

Okay, makes sense. In that case does it still make sense to merge certain things in automatically, while others needs to be threaded in manually? Or should we completely get rid of `mergeProjections` and manually merge the projection in everywhere?

No sure. Maybe it's fine for now. Just a thought.

Reply:
I tried it with spreading, but you have to spread so many objects it becomes hard to read and it's easy to forget spreading an object. Then I tried it with `merge`, but ran into the array merging behaviour. I'm pretty confident that `mergeProjection` is the best solution.